src/deploy/functions/runtimes/node/index.ts

```diff
@@ -1,16 +1,28 @@
 import { promisify } from "util";
 import * as fs from "fs";
 import * as path from "path";
+import * as portfinder from "portfinder";
+import * as spawn from "cross-spawn";
+import * as semver from "semver";
+import fetch from "node-fetch";
 
 import { FirebaseError } from "../../../../error";
 import { getRuntimeChoice } from "./parseRuntimeAndValidateSDK";
 import { logger } from "../../../../logger";
+import { previews } from "../../../../previews";
+import { logLabeledWarning } from "../../../../utils";
 import * as backend from "../../backend";
 import * as runtimes from "..";
 import * as validate from "./validate";
 import * as versioning from "./versioning";
 import * as parseTriggers from "./parseTriggers";
+import * as discovery from "../discovery";
 
+const MIN_FUNCTIONS_CONTROL_API_VERSION = "3.17.0";
```

Comment:
Be careful about dependency ordering here too. Don't release this CLI version until you _know_ 3.17.0 supports this endpoint and it's been bug bashed.